src/client/app/components/meters/EditMeterModalComponent.tsx

```diff
@@ -231,8 +236,19 @@ export default function EditMeterModalComponent(props: EditMeterModalComponentPr
 				// The input passed validation.
 				// GPS may have been updated so create updated state to submit.
 				const submitState = { ...state, gps: gps };
+				// The reading views need to be refreshed if going to/from no unit or
+				// to/from type quantity.
+				// The check does it by first seeing if the unit changed and, if so, it
+				// sees if either were non unit meaning it crossed since both cannot be no unit
+				// or the unit change to/from quantity.
+				const shouldRefreshReadingViews = (props.meter.unitId != state.unitId) &&
+					((props.meter.unitId == -99 || state.unitId == -99) ||
+						(unitState[props.meter.unitId].unitRepresent == UnitRepresentType.quantity
+							&& unitState[state.unitId].unitRepresent != UnitRepresentType.quantity) ||
+						(unitState[props.meter.unitId].unitRepresent != UnitRepresentType.quantity
+							&& unitState[state.unitId].unitRepresent == UnitRepresentType.quantity));
```

Comment:
This logic seems fairly complex, but after trying to simplify it a bit I found no shorter ways to do it. The comment is descriptive though so I think it is fine.

Reply:
Yes, I wanted to make it simpler but also found that all the tests were needed.